https://chromium.googlesource.com/chromiumos/platform2/+/45cccb17d4e0c375b4eb1729c2ed14278035363d

common-mk: Workaround gtest-config and gmock-config not being installed.

Upstream gtest-1.8.1 switches from autotools to cmake which drops
gtest-conf and gmock-conf. Therefore, it makes sense to provide the
linker flags directly to avoid rewriting the gtest ebuild.

BUG=chromium:940320
TEST=PreCQ passes

Change-Id: I20ff30e7800d922902dc15774ed73225b5377a43
Reviewed-on: https://chromium-review.googlesource.com/1532845
Commit-Ready: ChromeOS CL Exonerator Bot <chromiumos-cl-exonerator@appspot.gserviceaccount.com>
Tested-by: Allen Webb <allenwebb@google.com>
Reviewed-by: Mike Frysinger <vapier@chromium.org>
diff --git a/common-mk/BUILD.gn b/common-mk/BUILD.gn
index 1f535fece..509dab6bf 100644
--- a/common-mk/BUILD.gn
+++ b/common-mk/BUILD.gn
@@ -151,8 +151,6 @@ if (use.test || use.fuzzer) {
   # error from happening when ASAN is enabled.
   # TODO(crbug.com/887845): Remove this note after library order issue is resolved.
   config("test") {
-    ldflags = []
-
     # Don't worry about overlinking, ld.gold's --as-needed will
     # deal with that.
     # TODO(crbug.com/912432): Consider use {gtest,gmock}-config --ldflags.
@@ -162,17 +160,12 @@ if (use.test || use.fuzzer) {
     # -DNDEBUG, but the target built without, which causes a link error.
     # cf) BaseInitLoggingImpl definition in
     #     aosp/external/libchrome/base/logging.h
-    ldflags += exec_script("//common-mk/args_generator_wrapper.py",
-                           [
-                             "gtest-config",
-                             "--libs",
-                           ],
-                           "list lines")
-    ldflags += exec_script("//common-mk/args_generator_wrapper.py",
-                           [
-                             "gmock-config",
-                             "--libs",
-                           ],
-                           "list lines")
+    # gtest-config and gmock-config are not installed by gtest-1.8.1.
+    ldflags = [
+      "-lgmock",
+      "-lgtest",
+      "-pthread",
+      "-lpthread",
+    ]
   }
 }
diff --git a/common-mk/common_test.gypi b/common-mk/common_test.gypi
index e63f86624..14e9950bb 100644
--- a/common-mk/common_test.gypi
+++ b/common-mk/common_test.gypi
@@ -9,8 +9,8 @@
     'libraries+': [
       # Don't worry about overlinking, ld.gold's --as-needed will
       # deal with that.
-      '>!@(gtest-config --libs)',
-      '>!@(gmock-config --libs)',
+      # gtest-config and gmock-config are not installed by gtest-1.8.1.
+      '-lgmock', '-lgtest', '-pthread', '-lpthread',
     ],
   },
 }
